### ISSUE TITLE
provide log-scale option for depletion and F_std; and provide multi-yr averaging

### ISSUE THREAD

**Issue by k-doering-NOAA** [nmfs-stock-synthesis/stock-synthesis#89]
Imported from redmine, Issue [#83055](https://vlab.ncep.noaa.gov/redmine/issues/83055)
 Opened by @iantaylor-NOAA on 2020-09-21
Status when imported: Resolved
 
In an email conversation (Sept 16-17 with subject "Wish List for ss3diags"), Henning Winker requested logBratio and logFratio in derived quantities.

Rick proposed adding a option+10 input for the log alternatives for all options:

1 # Depletion basis:  denom is: 0=skip; 1=rel X*SPB0; 2=rel SPBmsy; 3=rel X*SPB_styr; 4=rel X*SPB_endyr
so options 11, 12, 13, 14 would be ln(options 1-4)):

However, for F_std_basis, this could be confusing as the 10s digit is already used for the multi-year trailing average. A 1 in the 10s digit wouldn't be used for the multi-year option since the 1-year average is already the default, but it wouldn't be possible to combine the multi-year average with the log scale option.

An alternative implementation would be to use negative values to choose the log scale for both the Depletion basis and the F std basis (e.g. -22 would calculate the log(F/FMSY) with a 2-year trailing average).

Ideally, r4ss could detect the presence of the log-scale output and convert the values used in the time series plots to standard units (where the uncertainty interval would be representative of a lognormal distribution which is a better approximation to MCMC estimates, hence the request from Henning).

**Comment by k-doering-NOAA** [comment]
Tagging @Henning-Winker for awareness, although this thread may be a bit hard to follow (and this suggestion was from a while back)!

Henning, I will try to test this out and provide a summary of how this option can be implemented; feel free to test out before then or wait until after I test (current test version of SS is available [as a github artifact](https://github.com/nmfs-stock-synthesis/stock-synthesis/suites/2617259771/artifacts/57344924) for now, but is only available for a few weeks, let me know if a new link is needed).

**Comment by k-doering-NOAA** [comment]
@Rick-Methot-NOAA I just tried this with simple, but changing the line
```
2 # Depletion basis:  denom is: 0=skip; 1=rel X*SPB0; 2=rel SPBmsy; 3=rel X*SPB_styr; 4=rel X*SPB_endyr
```
in the starter file to:
```
102 # Depletion basis:  denom is: 0=skip; 1=rel X*SPB0; 2=rel SPBmsy; 3=rel X*SPB_styr; 4=rel X*SPB_endyr
```
(I left this line in the starter, which maybe I need to change?)
```
0 # F_report_basis: 0=raw_F_report; 1=F/Fspr; 2=F/Fmsy ; 3=F/Fbtgt
```

I'm not sure this is working properly, as the output I see in echo input looks a little different:
```
102  depletion_basis as read; this is also known as Bratio and is a std quantity; has multi-yr and log(ratio) options
Parse into: depletion_log(ratio): 1 depletion_multi-yr: 0 depletion_basis: 102
1  depletion_level
4  SPR_reporting
3  F_reporting quantity, e.g. 3=sum(apical Fs)
0  F_std basis as read
Parse into: F_std_log(ratio): 0 F_std_multi: 0 F_std_basis: 0
For Kobe plot, set depletion_basis=2; depletion_level=1.0; F_reporting=your choose; F_std_basis=2
```

Here is how the labels look in ss_summary.sso:
```
#_F Basis= _abs_F;_with_F=sum(full_Fs)
```

```
#_Depletion Basis= ;log
Bratio_1973 10.8163 0.0277665
```

**Comment by k-doering-NOAA** [comment]
Ok, I ran a better comparison, where Bratio and Fratio were calculated on the nominal scale in 1 model, then on the log scale in the other. To reproduce this model, just change 2 lines in the starter file of simple; on the left is the log scale file, on the right is the nominal scale starter file
![2 files each with 2 lines of text highlighted, where 102 are the inuts for the log scale Bratio and F ratio and 2 are the inputs for the nominal scale values](https://user-images.githubusercontent.com/48930335/117363146-07c7bc80-ae71-11eb-9857-8f7a94ccda87.png)

I think some of these values are not lining up(i assume that taking the natural log of the nominal value could provide the value of the log scale point estimate, although perhaps there is also a correction that needs to be made?) - thanks to @Henning-Winker for suggesting this was the case.
Here are some of the Fs, where the values on the left are log scale and on the right are nominal scale: 
![image](https://user-images.githubusercontent.com/48930335/117363659-ba981a80-ae71-11eb-8126-14d5f410f3ee.png)

Here are some of the bratio values, log on left and nominal on right: 
![image](https://user-images.githubusercontent.com/48930335/117363718-cedc1780-ae71-11eb-8ee2-f10686904108.png)

**Comment by Rick-Methot-NOAA** [comment]
The code is really, really simple:
after calculating F_std in which it is divided by the reference point to
get a ratio,

  if(F_std_log==1) F_std = log(F_std);

**Comment by k-doering-NOAA** [comment]
That makes sense to me! Why don't these numbers line up then (or maybe I am miscalculating?)
Nominal scale:
#_F Basis= (F)/(Fmsy);_with_F=sum(full_Fs)
F_1972 0.0200349 0.00357247

Log scale: 
#_F Basis= ;_with_F=sum(full_Fs);log
F_1972 -6.16068 0.0409965

log(0.0200) = -3.912023
Maybe I am missing a correction factor?

**Comment by Rick-Methot-NOAA** [comment]
something strange.  I'll investigate and report back to you

**Comment by Rick-Methot-NOAA** [comment]
got it.

read value of 102 was not getting transformed to 2, so the switch statement
was not working right.  thanks for finding this problem

**Comment by k-doering-NOAA** [comment]
Good find! I wondered if that was the case with what the print statements in echoinput were saying. 

Let me know when a fix has been pushed and I can test out again.

**Comment by Rick-Methot-NOAA** [comment]
Fixed.  Revised code is in same branch that is adding se(smrybio).
Echoinput was displaying the problem as the input was parsed; I forgot to look there first.
Now echoinput shows:
182  depletion_basis as read; this is also known as Bratio and is a std quantity; has multi-yr and log(ratio) options
Parse into: depletion_log(ratio): 1 depletion_multi-yr: 8 depletion_basis: 2
 ........
132  F_std basis as read
Parse into: F_std_log(ratio): 1 F_std_multi: 3 F_std_basis: 2
For Kobe plot, set depletion_basis=2; depletion_level=1.0; F_reporting=your choose; F_std_basis=2
